app/core/utils.coffee

```diff
@@ -875,27 +875,91 @@ yearsSinceMonth = (start) ->
   return undefined unless start
   # Should probably review this logic, written quickly and haven't tested any edge cases
   if _.isString start
-    return undefined unless /\d{4}-\d{2}(-\d{2})?/.test start
+    return undefined unless /^\d{4}-\d{2}(-\d{2})?$/.test start
     if start.length is 7
       start = start + '-28'  # Assume near the end of the month, don't let timezones mess it up, skew younger in interpretation
     start = new Date(start)
   return undefined unless _.isDate start
   now = new Date()
   now.getFullYear() - start.getFullYear() + (now.getMonth() - start.getMonth()) / 12
 
+daysBetween = (date1, date2) ->
+  differenceInTime = date1.getTime() - date2.getTime()
+  differenceInDays = Math.round(differenceInTime / (1000 * 3600 * 24))
+  return differenceInDays
+
 # Keep in sync with the copy in background-processor
 ageBrackets = [
-  {slug: '0-11', max: 11.33}
-  {slug: '11-14', max: 14.33}
-  {slug: '14-18', max: 18.99}
+  {slug: '0-11', max: 4140 / 365.5}
+  {slug: '11-14', max: 5236 / 365.5}
+  {slug: '14-18', max: 6940 / 365.5}
   {slug: 'open', max: 9001}
 ]
 
+seasons = [
+  {
+    name: 'Season 1',
+    start:'01-01',
+    end: '04-30',
+    ageBrackets: [
+      {slug: '0-11', max: 4140 / 365.5}
+      {slug: '11-14', max: 5236 / 365.5}
+      {slug: '14-18', max: 6940 / 365.5}
+      {slug: 'open', max: 9001}
+    ]
+  }
+  {
+    name: 'Season 2',
+    start:'05-01',
+    end: '08-31',
+    ageBrackets: [
+      {slug: '0-11', max: 4260 / 365.5}
+      {slug: '11-14', max: 5356 / 365.5}
+      {slug: '14-18', max: 6940 / 365.5}
+      {slug: 'open', max: 9001}
+    ]
+  }
+  {
+    name: 'Season 3',
+    start:'09-01',
+    end: '12-31',
+    ageBrackets: [
+      {slug: '0-11', max: 4018 / 365.5}
+      {slug: '11-14', max: 5114 / 365.5}
+      {slug: '14-18', max: 6940 / 365.5}
+      {slug: 'open', max: 9001}
+    ]
+  }
+]
+
+currentSeason = () ->
+  now = new Date()
+  year = now.getFullYear()
+  return seasons.find((season) -> now <= new Date("#{year}-#{season.end}"))
+
+isInBracket = (age,bracket,season) ->
+  age = Math.round(age * 365.5)
+  bracketMax = bracket.max * 365.5
+  now = new Date()
+  seasonStart = new Date("#{now.getFullYear()}-#{season.start}")
+  seasonEnd = new Date("#{now.getFullYear()}-#{season.end}")
+  daysLeftFromSeason = daysBetween(now, seasonEnd) * -1
+  daysElapsedFromSeason = daysBetween(now, seasonStart)
+
+  seasonLength = daysLeftFromSeason + daysElapsedFromSeason
+  daysOlderThanBracketMax = age - bracketMax
+
+  willAgeIn = daysOlderThanBracketMax < seasonLength
+  notAgedIn = daysOlderThanBracketMax > daysElapsedFromSeason
+
+  return age < bracketMax || ( willAgeIn && notAgedIn)
```

Comment:
i can't really understand this `(willAgeIn && not AgedIn)` . can you give me a example case ?

Reply:
@smallst Sorry, the logic was not really clear, I've just pushed another commit with more meaningful variable names.

The `bracketMax` is the max age at the _beginning_ of the season.
Which means, the player can be older than the `bracketMax` if he's older only by `seasonLength` and not more. (`willAgeInToCurrentBracket`)
But if we're already in the middle of the season and the player already old enough for the next season, he'll go into the next one (`agedInToNextBracket`)

It means the player fits the actual season if `age < bracketMax || ( willAgeInToCurrentBracket && !agedInToNextBracket)`

I hope the logic is right and clear now. :-/

Reply:
@adamkecskes sorry to feel confusion again. got clear by `the bracketMax is the max age at the beginning of the season` it's good! but then:
1. at the begining of the season the logic should be `age < bracketMax` only right?
2. so time flies, there is `daysElapsedFromSeason` now, should the logic be `age - daysElapsedFromSeason < bracketMax` only? then that will be `daysOlderThanBracketMax < dasysElaspsedFromSeason` ?

Reply:
@smallst 
Yes, the logic is `[xxxxxx____xxxxx]` because during the `____` days, the player should go to the next season.
If the player is 11.4 yrs old, and we're near the beginning of the season, he'll go into `11-14` bracket, but if we're near to the end of the season, he will go into `0-11` bracket.